sql/rpl_gtid_set.cc

```diff
@@ -304,6 +304,22 @@ void Gtid_set::clear_set_and_sid_map() {
   DBUG_VOID_RETURN;
 }
 
+void Gtid_set::remove(const std::vector<rpl_sidno> &sidnos) {
+  if (sid_lock != NULL) sid_lock->assert_some_wrlock();
+  cached_string_length = -1;
+  rpl_sidno max_sidno = get_max_sidno();
+  if (!max_sidno) return;
+  Free_intervals_lock lock(this);
+  lock.lock_if_not_locked();
+  for (auto sidno : sidnos) {
+    if (sidno >= 1 && sidno <= max_sidno) {
+      Interval_iterator ivit(this, sidno);
+      if (ivit.get()) ivit.remove(this);
+    }
+  }
+  lock.unlock_if_locked();
```

Comment:
Redundant: `unlock_if_locked` will be called automatically in `lock`'s destructor.